tool-plugins/vscode/pom.xml

```diff
@@ -69,6 +158,51 @@
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
+                <groupId>com.google.code.maven-replacer-plugin</groupId>
+                <artifactId>replacer</artifactId>
+                <version>1.5.3</version>
+                <executions>
+                    <execution>
+                        <id>replace-version</id>
+                        <phase>validate</phase>
+                        <goals>
+                            <goal>replace</goal>
+                        </goals>
+                    </execution>
+                </executions>
+                <configuration>
+                    <includes>
+                        <include>${basedir}/package.json</include>
+                    </includes>
+                    <replacements>
+                        <replacement>
+                            <token>"version": "(.*?)",</token>
+                            <value>"version": "${project.version}",</value>
+                        </replacement>
+                    </replacements>
+                </configuration>
+            </plugin>
         </plugins>
     </build>
+    <profiles>
+        <profile>
+            <id>platform-windows</id>
+            <activation>
+                <os>
+                    <family>windows</family>
+                </os>
+            </activation>
+            <properties>
+                <!-- Override the executable names for Windows -->
+                <npm.executable>npm.cmd</npm.executable>
+            </properties>
+        </profile>
+    </profiles>
+
+    <properties>
+        <npm.executable>npm</npm.executable>
+        <langserver.launcher.id>language-server-stdio-launcher</langserver.launcher.id>
+        <langserver.launcher.version>${project.version}</langserver.launcher.version>
+    </properties>
```

Comment:
Shouldn't we have all the properties in the root pom of the project?

Reply:
I think this property is not needed anymore. Since we will be depending on the `${project.version}` version of the langserver stdio launcher. Instead of this property I'll use `${project.version}` inside dependencies.

Reply:
The `langserver.launcer.id` is specific to a plugin. Some other editor plugin could depend on a different launcher. So it make sense that its here.
it does make sense to move `npm.executable` property to the root pom. I'll update.

Reply:
@Manuri addressed in a5c8359b1a26